morph_tool/marker_helper.py

```diff
@@ -0,0 +1,200 @@
+"""Module to read/write/plot morphology markers."""
+import yaml
+from pathlib import Path
+import numpy as np
+import logging
+
+logger = logging.getLogger(__name__)
+
+
+class MarkerSet:
+    """Container of several markers for a single morphology."""
+
+    def __init__(self, markers):
+        """Create list of markers.
+
+        Either a path to a yaml file, or a list of Marker objects.
+        """
+        if isinstance(markers, str):
+            with open(markers, "r") as f:
+                self._from_dicts(yaml.safe_load(f))
+        else:
+            morph_names = {marker.morph_name for marker in markers}
+            if len(morph_names) > 1:
+                raise Exception("Markers of different morphologies were provided.")
+            self.markers = markers
+            self._morph_name = morph_names.pop()
+            self.morph_path = markers[0].morph_path
+
+    def _from_dicts(self, markers):
+        """Load marker from dict."""
+        self.markers = [Marker(**marker) for marker in markers["markers"]]
+        self._morph_name = markers["morph_name"]
+        self.morph_path = markers["morph_path"]
+
+    @property
+    def morph_name(self):
+        """"""
+        return self._morph_name
+
+    def plot(self, filename="markers.html", with_plotly=True):
+        """Plot morphology with markers."""
+        if with_plotly:
+            from neurom import load_neuron
+            from neurom.geom import bounding_box
+            from plotly_helper.neuron_viewer import NeuronBuilder
+            from plotly_helper.object_creator import scatter, vector
+
+            neuron = load_neuron(self.morph_path)
+            builder = NeuronBuilder(neuron, "3d", line_width=4, title=f"{self.morph_name}")
+
+            for marker in self.markers:
+                data = np.array(marker._data)
+                if marker.type == "points":
+                    if len(np.shape(marker._data)) == 1:
+                        data = data[np.newaxis]
+                    builder.helper.add_data(
+                        {
+                            f"{marker._label}": scatter(
+                                data, name=f"{marker._label}", showlegend=True, **marker.plot_style
+                            )
+                        }
+                    )
+                elif marker.type == "line":
+                    point_x = np.array(marker._data[0])
+                    point_y = np.array(marker._data[1])
+
+                    bbox = bounding_box(neuron)
+                    bbox[0] -= np.array([10, 10, 10])
```

Comment:
haha yes!  This is a hack so that the bbox is a bit larger than the neuron. This plotting function is a mess, I was just trying to plot lines quickly. We may want to make it a separate module, to make sure we don't pull plot dependencies. Not sure how to do best, I was thinking of something like `dataframe.plot()`. So  it's easy to use.

Reply:
I think it is ok to let it here and import specific dependencies in the plot method.
My point was just to make as many values as possible customizable, especially the ones like a bbox whose relevancy depends on the extent of the plotted data.